include/dali/c_api.h

```diff
@@ -30,13 +30,11 @@ extern "C" {
  * @note Beware, the C API is just C-like API for handling some mangling issues and
  * it can throw exceptions.
  */
-typedef struct {
-  void *pipe;
-  void *ws;
-  void *batch_size_map;      /// @see batch_size_map_t
-  cudaStream_t copy_stream;  /// Stream to perform copy operations on
-  void *data_id_map;         /// @see data_id_map_t
-} daliPipelineHandle;
+typedef struct DALIPipeline *daliPipelineHandle;
+
+// Temporary workaround for backward compatibility
+// - in the long run, daliPipeline_t is going to be change to daliPipeline and passed by value
+typedef daliPipelineHandle *daliPipelineHandle_t;
```

Comment:
Could you add explicitly in the comment, which structure should be used? `daliPipelineHandle` or `daliPipelineHandle_t`? It's not clear for me here.

Reply:
I'll just remove it or make it private (in the CC file). It will only confuse the users and won't make the transition to the future API any easier.